src/v/debug_bundle/probe.cc

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "debug_bundle/probe.h"
+
+#include "config/configuration.h"
+#include "debug_bundle/types.h"
+#include "metrics/prometheus_sanitize.h"
+
+#include <seastar/core/metrics.hh>
+#include <seastar/core/metrics_registration.hh>
+#include <seastar/core/shard_id.hh>
+
+using namespace std::chrono_literals;
+
+namespace debug_bundle {
+void probe::setup_metrics() {
+    namespace sm = ss::metrics;
+    auto setup_common = [this]<typename MetricDef>() {
+        std::vector<MetricDef> defs;
+        if (ss::this_shard_id() == service_shard) {
```

Comment:
Should this check be moved? What's the implication of registering an empty set of metrics?

I guess we don't actually get here on other shards, so this is just belt and braces.

Reply:
yeah I think it makes sense to move this outside of this lambda